tests/compute/test_specialization.py

```diff
@@ -47,18 +47,18 @@ def apply_func(nodes):
         assert F.allclose(v2, v3)
         # update all with edge weights
         v1 = g.ndata[fld]
-        g.update_all(fn.src_mul_edge(src=fld, edge='e1', out='m'),
-                fn.sum(msg='m', out=fld), apply_func)
-        v2 = g.ndata[fld]
-        g.set_n_repr({fld : v1})
-        g.update_all(fn.src_mul_edge(src=fld, edge='e2', out='m'),
-                fn.sum(msg='m', out=fld), apply_func)
-        v3 = g.ndata[fld]
+        if len(g.ndata[fld].shape) == 1:
```

Comment:
it seems the new builtin doesn't support some of the cases?

Reply:
DGL should be able to handle all f1*e1, f1*e2, f2*e1, f2*e2.